.github/workflows/test-action.yml

```diff
@@ -0,0 +1,118 @@
+name: Test GitHub Action
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    types: [ opened, synchronize, reopened, ready_for_review ]
+
+jobs:
+  test-defaults:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Test GitHub Action
+        uses: ./
+        with:
+          filepath: test/asyncapi.yml
+      - name: Assert GitHub Action
+        run: |
+          echo "Asserting GitHub Action"
+          if [ -f "./output/asyncapi.md" ]; then
+            echo "Files exist"
+          else
+            echo "Files do not exist:- ./output/asyncapi.md"
+            echo "Action failed"
+            exit 1
+          fi
+
+  test-validate-success:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Test GitHub Action
+        uses: ./
+        with:
+          filepath: test/asyncapi.yml
+          command: validate
+
+  test-custom-command:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Test GitHub Action
+        uses: ./
+        with:
+          custom_command: "generate models typescript ./test/asyncapi.yml -o ./output"
+      - name: Assert GitHub Action
+        run: |
+          echo "Asserting GitHub Action"
+          if [ -f "./output/AnonymousSchema_1.ts" ]; then
+            echo "Models have been generated"
+          else
+            echo "Models have not been generated"
+            echo "Action failed"
+            exit 1
+          fi
+  
+  test-custom-output:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Test GitHub Action
+        uses: ./
+        with:
+          filepath: test/asyncapi.yml
+          output: custom-output
+      - name: Assert GitHub Action
+        run: |
+          echo "Asserting GitHub Action"
+          if [ -f "./custom-output/asyncapi.md" ]; then
+            echo "Files exist"
+          else
+            echo "Files do not exist:- ./custom-output/asyncapi.md"
+            echo "Action failed"
+            exit 1
+          fi
+  
+  test-file-not-found:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Test GitHub Action
+        id: test
+        uses: ./
+        with:
+          filepath: non_existent_file.yml
+        continue-on-error: true
+      - name: Check for failure
+        run: if [ "${{ steps.test.outcome }}" == "success" ]; then exit 1; fi
```

Comment:
please echo some useful comment, that all is good as error was triggered

shouldn't you create here some file? like some `success` file` and then in another step check if file was created, then it means job successfully failed as expected - if it would not fail on `Test GitHub Action` then file would never be created 🤔 and job could fail

Reply:
I don't see the point of making a file as on error the outcome is set to failed, but step continues due to `continue-on-error`.


> As seen in docs :- 
> The result of a completed step before continue-on-error is applied. Possible values are success, failure, cancelled, or skipped. When a continue-on-error step fails, the outcome is failure, but the final conclusion is success.

Reply:
Will add the comment.